Advanced-Recruit-Training/src/BatteryStateOfChargeService.h

```diff
@@ -25,4 +26,10 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Add your necessary variables here
+    double current_;
+    double totalAmpHoursUsed_;
+    bool checkStart_;
+    double timeDiff_;
+    QTime time_;
```

Comment:
Try and be a bit more specific with these naming, ie is this the time of the "Present time" data point, or the time of the "Previous" data point? Same goes for the current.